File: Java/Map-2/wordLen.java
import java.util.HashMap;
import java.util.Map;

public class map2 {

	// given an array of strings, return a Map<String, Integer> containing a key for every different string in the array, and the value is that string's length
	public Map<String, Integer> wordLen(String[] str) {
		
		Map<String, Integer> map = new HashMap<String, Integer>();
		
		for (int i = 0; i < str.length; i++) {
			map.put(str[i], str[i].length());
		}
		
		return map;
		
	}
	
	public static void main(String[] args) {
		
		map2 o = new map2();
		
		String[] s1 = {"a", "bb", "a", "bb"};
		String[] s2 = {"this", "and", "that", "and"};
		String[] s3 = {"code", "code", "code", "bug"};
		
		System.out.println(o.wordLen(s1));
		System.out.println(o.wordLen(s2));
		System.out.println(o.wordLen(s3));
		

	}

}

File: Python/Map-2/word_len.py

# given an array of strings, return a Map<String, Integer> containing a key for every different string in the array, and the value is that string's length
def word_len(str):
    dict = {}

    for x in str:
        dict[x] = len(x)

    return dict

print(word_len(['a', 'bb', 'a', 'bb']))
print(word_len(['this', 'and', 'that', 'and']))
print(word_len(['code', 'code', 'code', 'bug']))
